db_sql.go

```diff
@@ -133,17 +150,39 @@ func (x *sqlAuthenticationDB) ResetPasswordFinish(identity string, token string,
 	}
 }
 
-func (x *sqlAuthenticationDB) CreateIdentity(identity, password string) error {
+func (x *sqlUserStoreDB) CreateIdentity(identity, password string) error {
 	hash, ehash := computeAuthausHash(password)
 	if ehash != nil {
 		return ehash
 	}
 
+	//insert into user store
 	if tx, etx := x.db.Begin(); etx == nil {
-		if _, ecreate := tx.Exec(`INSERT INTO authuser (identity, password) VALUES ($1, $2)`, identity, hash); ecreate == nil {
+		if _, ecreate := tx.Exec(`INSERT INTO userstore (identity) VALUES ($1)`, identity); ecreate != nil {
+			//fmt.Printf("CreateIdentity failed because: %v", ecreate)
```

Comment:
Everything inside this function must be wrapped inside a single transaction, and if any step fails, then the whole transaction must roll back. There is zero point in wrapping just the `INSERT INTO userstore` inside a transaction, because single statements are by definition always wrapped inside a transaction.